libraries/bsg_manycore_config.cpp

```diff
@@ -130,21 +130,28 @@ int hb_mc_config_init(const hb_mc_config_raw_t raw[HB_MC_CONFIG_MAX],
         config->vcore_dimensions.y = idx;
 
         idx = raw[HB_MC_CONFIG_DEVICE_HOST_INTF_COORD_X];
-        if ((idx < HB_MC_COORDINATE_MIN) || (idx > config->vcore_dimensions.x)){
-                bsg_pr_err("%s: Invalid Host Interface index X %d: %s\n",
-                           __func__, idx, error_init_help);
-                return HB_MC_INVALID;
-        }
+        // if ((idx < HB_MC_COORDINATE_MIN) || (idx > config->vcore_dimensions.x)){
+        //         bsg_pr_err("%s: Invalid Host Interface index X %d: %s\n",
+        //                    __func__, idx, error_init_help);
+        //         return HB_MC_INVALID;
+        // }
         config->host_interface.x = idx;
 
         idx = raw[HB_MC_CONFIG_DEVICE_HOST_INTF_COORD_Y];
-        if ((idx < HB_MC_COORDINATE_MIN) || (idx > config->vcore_dimensions.y)){
-                bsg_pr_err("%s: Invalid Host Interface index Y %d: %s\n",
-                           __func__, idx, error_init_help);
-                return HB_MC_INVALID;
-        }
+        // if ((idx < HB_MC_COORDINATE_MIN) || (idx > config->vcore_dimensions.y)){
+        //         bsg_pr_err("%s: Invalid Host Interface index Y %d: %s\n",
+        //                    __func__, idx, error_init_help);
+        //         return HB_MC_INVALID;
+        // }
         config->host_interface.y = idx;
 
+
+        idx = raw[HB_MC_CONFIG_NOC_COORD_X_WIDTH];
+        config->noc_coord_width.x = idx;
+
+        idx = raw[HB_MC_CONFIG_NOC_COORD_Y_WIDTH];
+        config->noc_coord_width.y = idx;
+
```

Comment:
Check validity?

Reply:
Tell me what "validity" means and I'll add the check.

Reply:
If it means a hardcoded number, we'll just be right back here next tape out. We'll also need separate versions of the library for each chip.

Reply:
I'm fine with just checking nonzero and not all 1's. And maybe less than say 32. All of those values are wild error conditions

Reply:
at that point why bother checking at all?

Reply:
A lot of these are smoke tests - like if we mapped an AXI platform incorrectly or the ROM is the wrong size.